### PR TITLE
feat(2087): Cache Clear feature implementation

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context
Cache cleanup buttons in Pipeline settings page, does not clear the cache when using disk cache.

## Objective
This PR implements a cache clear mechanism for disk based strategy by sending a message to the rabbitmq consumer

## References
https://github.com/screwdriver-cd/screwdriver/issues/2087

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/worker/lib/jobs.js [hunk 1]
```diff
@@ -197,6 +212,38 @@ function stop(buildConfig) {
...
+        .then(() => {
+            if (queueName) {
+                return pushToRabbitMq({ job: 'clear', cacheConfig }, queueName, id);
+            }
+
+            if (buildClusters) {
+                return Promise.all(
+                    buildClusters.map(cluster => pushToRabbitMq({ job: 'clear', cacheConfig }, cluster, id))
+                );
+            }
```
Comment (DekusDenial) [inline plugins/worker/lib/jobs.js:241]: double publishing? is the latter just to catch all?

File: plugins/worker/lib/CacheFilter.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     * @method constructor
+     */
+    constructor(worker, func, queue, job, args, options) {
+        super(worker, func, queue, job, args, options);
+
+        this.name = 'CacheFilter';
+    }
+
+    /**
+     * Checks if the job belongs to this worker
```
Comment (DekusDenial) [inline plugins/worker/lib/CacheFilter.js:17]: misleading description from c&p right?

Comment (pritamstyz4ever) [reply plugins/worker/lib/CacheFilter.js:17]: yup, fixing!!

File: plugins/worker/lib/jobs.js (hunk 1 above)
Comment (pritamstyz4ever) [reply plugins/worker/lib/jobs.js:241]: if queueName os not present, we publish to all existing rabbitmq queue's

Comment (DekusDenial) [reply plugins/worker/lib/jobs.js:241]: Sorry didn’t see the return there

Comment (parthasl) [reply plugins/worker/lib/jobs.js:241]: by default, it should push to all queues

Comment (DekusDenial) [reply plugins/worker/lib/jobs.js:241]: @pritamstyz4ever maybe all is better in case people switch cluster back and forth...

**Pull request merged**